### ISSUE TITLE
Custom subdomains stopped being created for new branches

### ISSUE THREAD

**Issue by hinaaman** [aws-amplify/amplify-hosting#3774]
### Before opening, please confirm:

- [X] I have checked the [custom domains troubleshooting guide](https://docs.aws.amazon.com/amplify/latest/userguide/custom-domain-troubleshoot-guide.html).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d1h6e93qbxs4wx

### Region

us-east-1

### Step

Preview url with custom domain

### Status

Fails to create

### Issue or question

We use custom subdomains for our preview builds. E.g.; `https://pr-1111.customdomain.dev`. These are auto-generated for each new branch. As of yesterday evening, the preview builds are defaulting to amplify urls. E.g; `https://pr-1111.cccc.amplifyapp.com`. Nothing has changed on our side. Settings seem fine and if we manually add a subdomain from Domain Management page, it works.

**Comment by Jay2113** [comment]
Hi @hinaaman 👋 , thanks for raising this!

When a PR is raised on the git repository, Amplify will create a new preview branch in the Amplify console which will be accessible via your custom domain URL such as `pr-{AWS_PULL_REQUEST_ID}.customdomain.dev`. This functionality will only work if you have enabled the workflow outlined [here](https://docs.aws.amazon.com/amplify/latest/userguide/pr-previews.html#web-preview-access-on-subdomains).

That being said, can you please further elaborate on the behavior that you are observing? When a preview build is created in the console, is the URL only accessible using the `amplifyapp.com` subdomain? Can you provide the affected branch names?

**Comment by hinaaman** [comment]
Hi @Jay2113,

Thanks for looking into this.

![Screen Shot 2023-10-27 at 10 39 42 AM](https://github.com/aws-amplify/amplify-hosting/assets/5608702/c6a43fc8-2e83-4f01-8ca3-17e086c8c337)

In the above screenshot, we see a list of recent pull requests. Notice how the Preview URL changed for recent PRs. It is no longer our custom domain. We have been manually adding the subdomain under domain management > subdomain management page. 

Example branch name: `new-branch-for-testing`

**Comment by Jay2113** [comment]
@hinaaman Are you observing this behavior in the console for all new PR branches? Are you able to reproduce it actively or intermittently for some new PR branches?

**Comment by hinaaman** [comment]
@Jay2113 Seems consistent on the new PR branches. I'm able to actively reproduce it.

**Comment by Jay2113** [comment]
Thanks for your continued patience @hinaaman .

Between 10/23 9:00 AM PT, and 10/27 3:00 PM PT, we experienced errors creating subdomain associations for Pull Requests. The issue has been resolved and the service is operating normally.

You should now be unblocked and new PR's should reflect the appropriate preview URL.
